### PR TITLE
Retain for job and metricsCollector

### PR DESCRIPTION
**Pull request by hougangliu**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/572)

### CODE REVIEW

**Comment by hougangliu** [comment]
Fixes: #565

File: pkg/controller/v1alpha2/trial/trial_controller.go [hunk 1]
```diff
@@ -245,6 +240,9 @@ func (r *ReconcileTrial) reconcileJob(instance *trialsv1alpha2.Trial, desiredJob
 	err = r.Get(context.TODO(), types.NamespacedName{Name: desiredJob.GetName(), Namespace: desiredJob.GetNamespace()}, deployedJob)
 	if err != nil {
 		if errors.IsNotFound(err) {
+			if instance.IsCompleted()  {
```
Comment (johnugeorge) [inline pkg/controller/v1alpha2/trial/trial_controller.go]: Which condition is being targeted here?

Comment (hougangliu) [reply pkg/controller/v1alpha2/trial/trial_controller.go]:
```
func (trial *Trial) IsCompleted() bool {
	return trial.IsSucceeded() || trial.IsFailed() || trial.IsKilled()
}
```

Comment (johnugeorge) [reply pkg/controller/v1alpha2/trial/trial_controller.go]: Is it the case that Get returns a stale entry after trial is completed?

Comment (johnugeorge) [reply pkg/controller/v1alpha2/trial/trial_controller.go]: https://github.com/kubeflow/katib/blob/master/pkg/controller/v1alpha2/trial/trial_controller.go#L149 returns if instance is completed.

Comment (hougangliu) [reply pkg/controller/v1alpha2/trial/trial_controller.go]: do you mean add a comment like `IsCompleted() targets "IsSucceeded() || IsFailed() || IsKilled()"` everywhere IsCompleted() is called?

Comment (hougangliu) [reply pkg/controller/v1alpha2/trial/trial_controller.go]: Line 147~151 deleted in this PR

Comment (johnugeorge) [reply pkg/controller/v1alpha2/trial/trial_controller.go]: Sorry. I missed that change.  Can you explain  the difference? And what I meant, is to add a comment why this check is added.  Readers will find it easier to follow the code.

**Pull request merged**